example/main.dart

```diff
@@ -1,23 +1,73 @@
+import 'dart:convert';
+
 import 'package:dio/dio.dart';
 import 'package:http_mock_adapter/http_mock_adapter.dart';
+import 'package:flutter_test/flutter_test.dart';
 
 void main() async {
-  final dio = Dio();
-  final dioAdapter = DioAdapter();
+  // How to mock with DioAdapter
+  group('DioAdapter usage', () {
+    // Creating dio instance for mocking.
+    // For instance: you can use your own instance from injection and replace
+    // dio.httpClientAdapter with mocker DioAdapter
+    final dio = Dio();
+    final dioAdapter = DioAdapter();
+
+    dio.httpClientAdapter = dioAdapter;
+
+    const path = 'https://example.com';
+
+    test('Expects Dioadapter to mock the data', () async {
+      dioAdapter
+          .onGet(path)
+          .reply(200,
+              {'message': 'Successfully mocked GET!'}) // only use double quotes
+          .onPost(path)
+          .reply(200, {'message': 'Successfully mocked POST!'});
+      final getResponse = await dio.get(path);
+
+      // Making dio.get request on the path an expecting mocked response
+      expect(jsonEncode({'message': 'Successfully mocked GET!'}),
```

Comment:
I think `final getResponse = await dio.get(path);` should be written above the `expect` just as you have it inside Interceptor example.